client/me/pending-payments/pending-list-item.jsx

```diff
@@ -3,33 +3,94 @@
  * External dependencies
  */
 import React from 'react';
-import { localize } from 'i18n-calypso';
+import { localize, moment } from 'i18n-calypso';
 import Gridicon from 'gridicons';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
 
 /**
  * Internal dependencies
  */
 import Card from 'components/card';
 import Button from 'components/button';
+import { getSite, getSiteTitle, getSiteDomain } from 'state/sites/selectors';
+import PurchaseSiteHeader from '../purchases/purchases-site/header';
+import { purchaseType as getPurchaseType, getName } from 'lib/purchases';
+import { paymentMethodName } from 'lib/cart-values';
+import formatCurrency from 'lib/format-currency';
+
+export function PendingListItem( {
+	translate,
+	paymentType,
+	totalCost,
+	currency,
+	siteId,
+	siteTitle,
+	siteDomain,
+	dateCreated,
+	products,
+} ) {
+	let productName = '';
+	let purchaseType = '';
+
+	if ( products.length > 1 ) {
+		productName = 'Multiple Items';
+		purchaseType = 'Various';
+	} else if ( products[ 0 ] ) {
```

Comment:
Are the cases `products.length > 1` and `products[ 0 ]` in some way different?  When would the first kick in and not the second?

I don't have full context, but my first thought would be to ask, why wouldn't something like this work?

```
	const initialProduct = get( products, '[0]', {} );
	const productName = getName( initialProduct ) || '';
	const purchaseType = getPurchaseType( initialProduct ) || '';
```